optaplanner-wb-screens/optaplanner-wb-domain-editor/optaplanner-wb-domain-editor-backend/src/main/java/org/optaplanner/workbench/screens/domaineditor/backend/server/PlannerDomainHandler.java

```diff
@@ -66,15 +66,17 @@ public void setDefaultValues( DataObject dataObject, Map<String, Object> options
 
     @Override
     public void processDataObject( DataObject dataObject, DataModel dataModel ) {
-        boolean hasPlanningEntity = dataObject.getAnnotation( PlannerDomainAnnotations.PLANNING_ENTITY_ANNOTATION ) != null;
-        boolean hasComparatorDefinition = dataObject.getAnnotation( ComparatorDefinition.class.getName() ) != null;
-        if ( hasPlanningEntity && hasComparatorDefinition ) {
-            try {
-                ComparatorObject comparatorObject = plannerDataObjectEditorService.extractComparatorObject( dataObject, dataModel );
-                dataObject.getNestedClasses().clear();
-                dataObject.getNestedClasses().add( comparatorObject );
-            } catch ( Exception e ) {
-                logger.error( "Unable to extract comparator object from " + dataObject.getClassName() );
+        if ( dataModel != null ) {
+            boolean hasPlanningEntity = dataObject.getAnnotation( PlannerDomainAnnotations.PLANNING_ENTITY_ANNOTATION ) != null;
+            boolean hasComparatorDefinition = dataObject.getAnnotation( ComparatorDefinition.class.getName() ) != null;
+            if ( hasPlanningEntity && hasComparatorDefinition ) {
+                try {
+                    ComparatorObject comparatorObject = plannerDataObjectEditorService.extractComparatorObject( dataObject, dataModel );
+                    dataObject.getNestedClasses().clear();
+                    dataObject.getNestedClasses().add( comparatorObject );
+                } catch ( Exception e ) {
+                    logger.error( "Unable to extract comparator object from " + dataObject.getClassName() );
+                }
```

Comment:
This catch **eats the exception**. Eating the exception means that when a user in the ui (or a developer in code) does an action and it goes wrong, he doesn't see it. He doesn't get an error message (normally in the UI this is a popup dialog or top screen notification whatever is appropriate - but it shouldn't dissappear faster than a slow reader can read it). Worse, if he sees nothing, the user presumes that his action succeeded. He'll feel scammed once he finds out that what he though he told the computer to do, didn't actually happen.

Nobody reads the production logs. Not the user in the UI, nor the developer when coding.

Error handling should be handled in this way:

- Always let the error bubble up to the highest layer possible, usually the thread's exception handler (or the exception handler of the servlet thread and the swing event thread). There, it must be 1) logged and 2) shown to the user.
- It's ok to wrap it. This can be to turn a checked exception into an unchecked exception. But it can also be to provide a more helpful error message. But always chain the original exception as the cause! For example: `throw new IllegalStateException("Error when reading file (" + file.getPath + ").", ioException);` Otherwise the user wouldn't know the filename that caused the IOException.
- On the highest layer, it should be logged (the level might depend on the type, for example AuthenticationException due to a wrong password login usually gets a less important log level such as info instead of error or warn. Users get their password wrong all the time.
- On the highest layer, it must also be shown to the user, clearly in red. This is the only way to put those errors in our face during development. They must stick out like a soar thumb. In the short time this will annoy users, in the long term this will make errors disappear because we see them and our user see them too and can report them.